### ISSUE TITLE
Resque macro preventing dead worker exits

### ISSUE THREAD

**Issue by jwoodrow** [hirefire/hirefire-resource#64]
Hi 👋 

We've been having some issues for some time now relating to when hirefire downscales the dyno count. Basically anytime a dyno gets shutdown it has 30 seconds to finish what it was doing but our jobs tend to last more than 30 seconds. This leads to jobs being reported as failed anytime hirefire downscales the dyno count.

Is there any way for hirefire to shut down specific dynos based on a metric (maybe a custom `workers.working_count` or something) to avoid this issue altogether ? I don't know if this is event possible using heroku's APIs (shutting down `resque.3` for example when there are 5 dynos)

Or alternatively on the web interface I wouldn't mind also having a downscaling threshold (delays don't seem to cut it so I would like to potentially only downscale when the queue size is 0 for example)

**Comment by jwoodrow** [comment]
Hi @mrrooijen 

> There is no way to specify which dyno to shutdown. This is indeed a limitation of Heroku's API. It always shuts down the highest worker number (worker.N). For example, if you have 10 workers and you want to scale down to 5, it will shutdown worker.(10,9,8,7,6).

I figured this might be the case 😞 

> You can do this by disabling the 'Decrementable' option, located in the worker dyno manager settings.

We've already disabled this option everywhere but it seems like it's not behaving that way for us 🤔 I'm not excluding the possibility that these our happening by our fault but the dead worker exit should only happen when workers are terminated 🤔

**Comment by mrrooijen** [comment]
I'm not aware of any issues on our end regarding the Decrementable option. Perhaps these dead exits are also caused by events such as app deployments, version rollbacks, environment variable changes, or daily automated restarts? In any case, since there are various (non-HireFire related) events that trigger shutdowns, it's always worth considering making jobs that take longer than 30 seconds idempotent or splitting them into smaller jobs that take less than 30 seconds each, to ensure that none of these events cause any issues.

**Comment by jwoodrow** [comment]
I did some slight changes on our end and it appears these dirty exits might have come from a cleanup scheduled job that was being run every minute (we've been having a zombie worker problem for a couple of weeks now and I got fed up and decided to to a cleanup every minute of any worker no longer associated with a running dyno). I changed this to be running every 30 minutes. We still have some zombie workers but this allows jobs to run with much less hiccups.

Seems like this really wasn't related to hirefire 👋
